### PR TITLE
Notfound

### PR DESCRIPTION
**Pull request by martindurant**

### CODE REVIEW

File: s3fs/core.py [hunk 1]
```diff
@@ -185,7 +190,7 @@ def ls(self, path, detail=False):
         try:
             files = self._ls(path)
         except ClientError:
-            files = []
+            raise FileNotFoundError(path)
```
Comment (mrocklin) [inline s3fs/core.py]: Things like this could use tests, ideally in the same commit where the functionality was added.

File: s3fs/core.py [hunk 2]
```diff
@@ -193,7 +198,7 @@ def ls(self, path, detail=False):
                 try:
                     files = [self.info(path)]
                 except (OSError, IOError, ClientError):
-                    files = []
+                    []
```
Comment (mrocklin) [inline s3fs/core.py]: ?

**Comment by martindurant** [comment]
Actually, I'll just get rid of all of this for now - clearly not turning out quality.

File: s3fs/core.py (hunk 2 above)
Comment (mrocklin) [inline s3fs/core.py]: The fact that this change was made and tests still passed is very concerning.

**Pull request closed without merging**